[flang][docs] Documented IS_CONTIGUOUS() extension for constant arrays (#200451)
llvm/llvm-project: flang/docs/Extensions.md

  with bounds remapping.
* The `CONTIGUOUS` attribute can be redundantly applied to simply
  contiguous objects, including scalars, with a portability warning.
* `IS_CONTIGUOUS` is always true when its argument is a named constant
  (`PARAMETER`) or a subobject of a named constant, even when the standard's
  rules in F2023 8.5.7 would otherwise classify the subobject as non-contiguous
  (for example, a strided section like `a(::2)`).  This happens because Flang
  constant-folds an array section of a named constant into a fresh constant array
  containing only the selected elements, and that fresh array is trivially
  contiguous.  When `-pedantic` is enabled, Flang emits the

>   `-Wconstant-is-contiguous` warningat each such call.  For example:

Reviewer: ```suggestion
  `-Wconstant-is-contiguous` warning at each such call.  For example:
```

> ```fortran
> integer, parameter :: a(5) = [1,2,3,4,5]
> print *, is_contiguous(a(::2))                   ! prints T in Flang
> ```

Reviewer: IIRC, if this is indented, it will be aligned with the text of the paragraph. Otherwise, it will be indented to the left relative to the text. I think aligning this with the paragraph would be preferable, but if you don't agree, feel free to leave it as it is.

```suggestion
  ```fortran
  integer, parameter :: a(5) = [1,2,3,4,5]
  print *, is_contiguous(a(::2))                   ! prints T in Flang
  ```
```

>   The other compilers may report `a(::2)` as non-contiguous in this case.

Reviewer: ```suggestion
  Other compilers may report `a(::2)` as non-contiguous.
```
